v03_pipeline/lib/tasks/reference_data/updated_reference_dataset_collection_task.py

```diff
@@ -0,0 +1,206 @@
+import os
+from datetime import datetime
+from functools import cached_property
+from types import FunctionType
+
+import hail as hl
+import luigi
+import pytz
+
+from v03_pipeline.lib.model import ReferenceDatasetCollection
+from v03_pipeline.lib.paths import valid_reference_dataset_collection_path
+from v03_pipeline.lib.reference_data.config import CONFIG
+from v03_pipeline.lib.tasks.base.base_update_task import BaseUpdateTask
+from v03_pipeline.lib.tasks.files import GCSorLocalTarget
+
+# NOTE: this is intended to replace the code in v03_pipeline/bin/write_combined_reference_ht.py, v03_pipeline/bin/write_combined_interval_reference_ht.py, and combine.py
+
+class UpdatedReferenceDatasetCollectionTask(BaseUpdateTask):
+    reference_dataset_collection = luigi.EnumParameter(enum=ReferenceDatasetCollection)
+    dataset = luigi.OptionalStrParameter()
+
+    @cached_property
```

Comment:
I think this is probably not necessary here, though the idea is correct.  iirc the place I used this recently involved a hail update of some kind... this is just a string path join under the hood and I don't think needs to be cached.